llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -123,14 +123,18 @@ def __init__(
         self._os_async_client = os_async_client or self._get_async_opensearch_client(
             self._endpoint, **kwargs
         )
-        self._os_version = self._get_opensearch_version()
-        self._efficient_filtering_enabled = self._is_efficient_filtering_enabled(
-            self._os_version
-        )
+        self._efficient_filtering_enabled = self._is_efficient_filtering_enabled()
         not_found_error = self._import_not_found_error()
 
         try:
             self._os_client.indices.get(index=self._index)
+        except TypeError:
+            # Probably using async so switch to async client
+            try:
+                asyncio.run(self._os_async_client.indices.get(index=self._index))
```

Comment:
this is a little error prone -- a little safer to do

```
from llama_index.core.async_utils import asyncio_run

asyncio_run(...)
```

This will make sure any existing async loop is used

Reply:
I added this